### PR TITLE
[docs] Update CONTRIBUTION guideline

### PR DESCRIPTION
**Pull request by leonjia0112**
Add step to pre-PR content that contributor needs to follow before submitting a PR.

### CODE REVIEW

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -71,6 +71,9 @@ git tools.
   - Format your code and fix all the spelling/grammatical mistakes.
   - Limit the column length of the code and the comments within the code to 120 characters.
   - Error messages within the code should be limited to a single line.
+  - Document the details of your PR and put a link inside your PR comment
```
Comment (aravindhp) [inline CONTRIBUTION.md]: Every PR needn't require documentation. Please replace with `Update the documentation if your PR is introducing or changing user facing functionality.`

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -71,6 +71,9 @@ git tools.
...
+  - Make sure your code compiles on your local machine
```
Comment (aravindhp) [inline CONTRIBUTION.md]: Please replace with `Ensure each PR commit compiles and all unit and e2e tests pass your local machine.`

File: CONTRIBUTION.md [hunk 3]
```diff
@@ -71,6 +71,9 @@ git tools.
...
+  - Run and make sure it passes all unit testing and e2e testing on your local machine
```
Comment (aravindhp) [inline CONTRIBUTION.md]: You can remove this bullet.

File: CONTRIBUTION.md (hunk 2 above)
Comment (leonjia0112) [reply CONTRIBUTION.md]: Is it better if I say `Ensure each PR commit compiles and all unit and e2e tests pass on your local machine`.

Comment (aravindhp) [reply CONTRIBUTION.md]: Yes, I missed a `on`

File: CONTRIBUTION.md [hunk 4]
```diff
@@ -71,6 +71,8 @@ git tools.
   - Format your code and fix all the spelling/grammatical mistakes.
   - Limit the column length of the code and the comments within the code to 120 characters.
   - Error messages within the code should be limited to a single line.
+  - Update the documentation if your PR is introducing or changing user facing functionality
+  - Ensure each PR commit compiles and all unit and e2e tests pass on your local machine
```
Comment (aravindhp) [inline CONTRIBUTION.md]: Both your statements are missing `.` at the end. Please be consistent with existing documentation.

Comment (leonjia0112) [reply CONTRIBUTION.md]: My bad, didn't notice that.

File: CONTRIBUTION.md [hunk 5]
```diff
@@ -71,6 +71,8 @@ git tools.
   - Format your code and fix all the spelling/grammatical mistakes.
   - Limit the column length of the code and the comments within the code to 120 characters.
   - Error messages within the code should be limited to a single line.
+  - Update the documentation if your PR is introducing or changing user facing functionality.
+  - Ensure each PR commit compiles and all unit and e2e tests pass on your local machine.
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTION.md]: Not sure, if we want to go this route

Comment (aravindhp) [reply CONTRIBUTION.md]: Which route? e2e or compile? I definitely want each commit to compile.

Comment (ravisantoshgudimetla) [reply CONTRIBUTION.md]: e2e tests pass - say if we've a vendor commit and the changes it brings in other commit

Comment (aravindhp) [reply CONTRIBUTION.md]: I am fine having an exception when we are introducing vendor commits. So we can add a line mentioning that. But in general I don't want to see commits that don't compile.

Comment (leonjia0112) [reply CONTRIBUTION.md]: Do you want to remove `e2e test` or having another bullet point saying that ` pass e2e test is optional for vendor commit but good to have`?

Comment (aravindhp) [reply CONTRIBUTION.md]: There should be a sub bullet stating that PRs that have vendor commits are an exception to the rule.

**Comment by ravisantoshgudimetla** [comment]
Trying  the magic incantation :)

File: CONTRIBUTION.md (hunk 5 above)
Comment (leonjia0112) [reply CONTRIBUTION.md]: Sub bullet added.

**Pull request merged**